account_financial_report/wizard/general_ledger_wizard.py

```diff
@@ -80,15 +85,45 @@ class GeneralLedgerReportWizard(models.TransientModel):
         string='Show foreign currency',
         help='Display foreign currency for move lines, unless '
              'account currency is not setup through chart of accounts '
-             'will display initial and final balance in that currency.'
+             'will display initial and final balance in that currency.',
+        default=lambda self: self._default_foreign_currency(),
     )
 
+    def _init_date_from(self):
+        """set start date to begin of current year if fiscal year running"""
+        today = fields.Date.context_today(self)
+        cur_month = fields.Date.from_string(today).month
+        cur_day = fields.Date.from_string(today).day
+        last_fsc_month = self.env.user.company_id.fiscalyear_last_month
+        last_fsc_day = self.env.user.company_id.fiscalyear_last_day
+
+        if cur_month < last_fsc_month \
+                or cur_month == last_fsc_month and cur_day <= last_fsc_day:
+            return time.strftime('%Y-01-01')
+
+    def _default_foreign_currency(self):
+        return self.env.user.has_group('base.group_multi_currency')
+
+    def _default_partners(self):
+        context = self.env.context
+
+        if context.get('active_ids') and context.get('active_model') \
+                == 'res.partner':
+            partner_ids = context['active_ids']
+            corp_partners = self.env['res.partner'].browse(partner_ids). \
+                filtered(lambda p: p.parent_id)
+
+            partner_ids = set(partner_ids) - set(corp_partners.ids)
+            partner_ids |= set(corp_partners.mapped('parent_id.id'))
```

Comment:
`partner_ids |= set(corp_partners.mapped('commercial_partner_id.id'))`

But it's better to work full method with recordsets:
```python
partners = self.env['res.partner'].browse(context['active_ids'])
corp_partners = partners.filtered('parent_id')
return partners - corp_partners + corp_partners.mapped('commercial_partner_id')
```

Reply:
Done